scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,367 @@
+    ...                                [-np.inf, 0],
+    ...                                [10, np.inf],
+    ...                                [2, 3]])
+    >>> print(data)
+    CensoredData(5 values: 2 not censored, 1 left-censored,
+    1 right-censored, 1 interval-censored)
+
+    A common case is to have a mix of uncensored observations and censored
+    observations that are all right-censored (or all left-censored). For
+    example, consider an experiment in which six devices are started at
+    various times and left running until they fail.  Assume that time is
+    measured in hours, and the experiment is stopped after 30 hours, even
+    if all the devices have not failed by that time.  We might end up with
+    data such as this::
+
+        Device  Start-time  Fail-time  Time-to-failure
+           1         0         13           13
+           2         2         24           22
+           3         5         22           17
+           4         8         23           15
+           5        10        ***          >20
+           6        12        ***          >18
+
+    Two of the devices had not failed when the experiment was stopped;
+    the observations of the time-to-failure for these two devices are
+    right-censored.  We can represent this data with
+
+    >>> data = CensoredData(x=[13, 22, 17, 15], right=[20, 18])
+    >>> print(data)
+    CensoredData(6 values: 4 not censored, 2 right-censored)
+
+    Alternatively, we can use the method `CensoredData.right_censored` to
+    create a representation of this data.  The time-to-failure observations
+    are put the list ``ttf``.  The ``censored`` list indicates which values
+    in ``ttf`` are censored.
+
+    >>> ttf = [13, 22, 17, 15, 20, 18]
+    >>> censored = [False, False, False, False, True, True]
+
+    Pass these lists to `CensoredData.right_censored` to create an
+    instance of `CensoredData`.
+
+    >>> data = CensoredData.right_censored(ttf, censored)
+    >>> print(data)
+    CensoredData(6 values: 4 not censored, 2 right-censored)
+    """
+
+    def __init__(self, x=None, *, left=None, right=None, intervals=None):
+        if x is None:
+            x = []
+        if left is None:
+            left = []
+        if right is None:
+            right = []
+        if intervals is None:
+            intervals = np.empty((0, 2))
+
+        _validate_1d(x, 'x')
+        _validate_1d(left, 'left')
+        _validate_1d(right, 'right')
+        x2, left2, right2, intervals2 = _validate_intervals(intervals)
+
+        self._x = np.concatenate((x, x2))
+        self._left = np.concatenate((left, left2))
+        self._right = np.concatenate((right, right2))
+        # Note that by construction, the private attribute _intervals
+        # will be a 2D array that contains only finite values representing
+        # intervals with nonzero but finite length.
+        self._intervals = intervals2
+
+    def __repr__(self):
```

Comment:
Fine. We could consider appending to the `__str__` to have the counts too.

Reply:
It looks like this distinction intentional. It is not uncommon for `__repr__` and `__str__` to have different amounts of detail.

```python3
import numpy as np; x = np.random.rand(10).astype(np.float32)

x
# array([0.8040919 , 0.20707697, 0.17506647, 0.40690997, 0.35903388,
#        0.69791615, 0.19661671, 0.29028037, 0.85509133, 0.09499677],
#       dtype=float32)

print(x)
# [0.8040919  0.20707697 0.17506647 0.40690997 0.35903388 0.69791615
#  0.19661671 0.29028037 0.85509133 0.09499677]
```
and for `__repr__` to include all the detail needed to create the object.

Reply:
Right 👍 I agree